op-node/rollup/derive/ecotone_upgrade_transactions.go

```diff
@@ -0,0 +1,132 @@
+)
+
+var (
+	// known address w/ zero txns
+	l1BlockDeployerAddress        = common.HexToAddress("0x4210000000000000000000000000000000000000")
+	gasPriceOracleDeployerAddress = common.HexToAddress("0x4210000000000000000000000000000000000001")
+
+	newL1BlockAddress        = crypto.CreateAddress(l1BlockDeployerAddress, 0)
+	newGasPriceOracleAddress = crypto.CreateAddress(gasPriceOracleDeployerAddress, 0)
+
+	deployL1BlockSource        = UpgradeDepositSource{Intent: "Ecotone: L1 Block Deployment"}
+	deployGasPriceOracleSource = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Deployment"}
+	updateL1BlockProxySource   = UpgradeDepositSource{Intent: "Ecotone: L1 Block Proxy Update"}
+	updateGasPriceOracleSource = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Proxy Update"}
+	enableEcotoneSource        = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Set Ecotone"}
+	beaconRootsSource          = UpgradeDepositSource{Intent: "Ecotone: beacon block roots contract deployment"}
+
+	eip4788From         = common.HexToAddress("0x0B799C86a49DEeb90402691F1041aa3AF2d3C875")
+	eip4788CreationData = common.Hex2Bytes("0x60618060095f395ff33373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500")
+)
+
+func EcotoneNetworkUpgradeTransactions() ([]hexutil.Bytes, error) {
+	upgradeTxns := make([]hexutil.Bytes, 0, 5)
+
+	deployL1BlockTransaction, err := types.NewTx(&types.DepositTx{
+		SourceHash:          deployL1BlockSource.SourceHash(),
+		From:                l1BlockDeployerAddress,
+		To:                  nil,
+		Value:               nil,
+		Gas:                 300_000,
+		IsSystemTransaction: false,
+		Data:                common.FromHex(bindings.L1BlockMetaData.Bin),
+	}).MarshalBinary()
+
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeTxns = append(upgradeTxns, deployL1BlockTransaction)
+
+	deployGasPriceOracle, err := types.NewTx(&types.DepositTx{
+		SourceHash:          deployGasPriceOracleSource.SourceHash(),
+		From:                gasPriceOracleDeployerAddress,
+		To:                  nil,
+		Value:               nil,
+		Gas:                 500_000,
+		IsSystemTransaction: false,
+		Data:                common.FromHex(bindings.GasPriceOracleMetaData.Bin),
+	}).MarshalBinary()
+
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeTxns = append(upgradeTxns, deployGasPriceOracle)
+
+	proxyAbi, err := bindings.ProxyMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeL1BlockData, err := proxyAbi.Pack("upgradeTo", newL1BlockAddress)
+	if err != nil {
+		return nil, err
+	}
+	updateL1BlockProxy, err := types.NewTx(&types.DepositTx{
+		SourceHash:          updateL1BlockProxySource.SourceHash(),
+		From:                common.Address{},
+		To:                  &predeploys.L1BlockAddr,
+		Value:               nil,
+		Gas:                 200_000,
```

Comment:
The gas values are underspecified still

Reply:
https://github.com/ethereum-optimism/client-pod/issues/421

Reply:
The non-dencun gas amounts are tracked by the issue, the gas of the dencun 4788 tx is a specific value as defined in the EIP. Resolving this, for the purposes of this PR.